allennlp/commands/fine_tune.py

```diff
@@ -159,23 +171,24 @@ def fine_tune_model(model: Model,
     vocabulary_params = params.pop('vocabulary', {})
     if vocabulary_params.get('directory_path', None):
         logger.warning("You passed `directory_path` in parameters for the vocabulary in "
-                       "your configuration file, but it will be ignored. "
-                       "Vocabulary from the saved model will be extended with current data.")
+                       "your configuration file, but it will be ignored. ")
 
     all_datasets = datasets_from_params(params)
 
-    datasets_for_vocab_creation = set(params.pop("datasets_for_vocab_creation", all_datasets))
+    if extend_vocab:
+        datasets_for_vocab_creation = set(params.pop("datasets_for_vocab_creation", all_datasets))
+
+        for dataset in datasets_for_vocab_creation:
+            if dataset not in all_datasets:
+                raise ConfigurationError(f"invalid 'dataset_for_vocab_creation' {dataset}")
 
-    for dataset in datasets_for_vocab_creation:
-        if dataset not in all_datasets:
-            raise ConfigurationError(f"invalid 'dataset_for_vocab_creation' {dataset}")
+        logger.info("Extending model vocabulary using %s data.", ", ".join(datasets_for_vocab_creation))
+        vocab = model.vocab
```

Comment:
i.e., i think this should go outside the `if extend_vocab` so `vocab.save_to_files(os.path.join(serialization_dir, "vocabulary"))` works in all cases

Reply:
done